core/src/test/java/org/elasticsearch/cluster/service/ClusterServiceTests.java

```diff
@@ -382,21 +408,54 @@ public void execute() {
                     counter.incrementAndGet();
                 }
             }
+
+            @Override
+            public boolean equals(Object o) {
+                if (this == o) {
+                    return true;
+                }
+                if (o == null || getClass() != o.getClass()) {
+                    return false;
+                }
+                Task task = (Task) o;
+                return id == task.id;
+
+            }
+
+            @Override
+            public int hashCode() {
+                return id;
+            }
+
+            @Override
+            public String toString() {
+                return Integer.toString(id);
+            }
         }
 
-        int numberOfThreads = randomIntBetween(2, 8);
-        int tasksSubmittedPerThread = randomIntBetween(1, 1024);
+        int numberOfThreads = randomIntBetween(2, 5);
+        int taskSubmissionsPerThread = randomIntBetween(1, 24);
```

Comment:
I tested with eight threads, and the test took < 20 ms to execute on my machine, so I definitely think there's room to keep the number of threads between two and eight, and I'm not sure that we need to go as low as twenty-four on the number of submissions?